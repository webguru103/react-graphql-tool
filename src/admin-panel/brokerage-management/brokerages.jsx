import React from 'react';
import { withRouter } from 'react-router';
import type { RouterHistory } from 'react-router';
import { withStyles } from '@material-ui/core/styles';
import { UnControlledTable } from '../../shared/table';
import { columns } from './columns';
import { withBrokerages } from '../api/admin.service';
import { compose } from '../../utility';
import { withAppUser } from '../../shared/authorization/userConsumer';
import styles from './brokerages.styles';
import type { BrokerageType } from '../../flowTypes';
import c from './constants';

type PropType = {
  history: RouterHistory,
  getBrokerages: () => Promise<Array<BrokerageType>>,
  classes: Object,
}

type StateType = {
  brokerages: Array<BrokerageType>,
  loading: boolean,
}

class Brokerages extends React.Component<PropType, StateType> {

  state = {
    brokerages: [],
    loading: false,
  };

  componentDidMount() {
    this.setState({ loading: true });
    this.props.getBrokerages()
      .then(brokerages => this.setState({ brokerages, loading: false }))
      .catch(() => this.setState({ loading: false })); /* TODO: handle error */
  }

  render() {
    return (
      <div className={this.props.classes.brokeragesTable}>
        <UnControlledTable
          data={this.state.brokerages}
          loading={this.state.loading}
          columns={columns}
          defaultPageSize={c.DEFAULT_PAGE_SIZE}
          onRowClick={row => this.props.history.push(`/admin/brokerage-management/${row.original.id}`)}
        />
      </div>
    );
  }

}

export default compose(
  withStyles(styles),
  withRouter,
  withAppUser,
  withBrokerages,
)(Brokerages);
