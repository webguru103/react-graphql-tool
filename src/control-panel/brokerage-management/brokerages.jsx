import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import AddBrokerage from './addBrokerage';
import ViewBrokerages from './viewBrokerages';
import styles from './brokerages.styles';

const BrokeragesC = ({ classes }: { classes: Object }) => (
  <React.Fragment>
    <div className={classes.nav}>
      <AddBrokerage />
    </div>
    <ViewBrokerages />
  </React.Fragment>
);

export default withStyles(styles, { withTheme: true })(BrokeragesC);
