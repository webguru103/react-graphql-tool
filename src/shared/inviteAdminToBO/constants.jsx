import React from 'react';
import { FormattedMessage } from 'react-intl';

export default {
  EMPTY_EMAIL: <FormattedMessage id="empty-email" defaultMessage="Please input an email" />,
  EMPTY_FIRST_NAME: <FormattedMessage id="empty-first-name" defaultMessage="Please input a first name" />,
  EMPTY_LAST_NAME: <FormattedMessage id="empty-last-name" defaultMessage="Please input a last name" />,
  INVALID_EMAIL: <FormattedMessage id="invalid-email" defaultMessage="Email is invalid" />,
  REQUEST_ERROR: <FormattedMessage id="request-error" defaultMessage="Sorry, there was an error processing your request!" />,
  EMPTY_BROKERAGE: <FormattedMessage id="empty-brokerage" defaultMessage="Please choose the brokerage from the dropdown list" />,
};
