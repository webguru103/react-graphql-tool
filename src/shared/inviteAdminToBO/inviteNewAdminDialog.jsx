import * as React from 'react';
import classNames from 'classnames';
import { Formik } from 'formik';
import Select from 'react-select';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormHelperText from '@material-ui/core/FormHelperText';
import { withStyles } from '@material-ui/core/styles';
import styles from './inviteNewAdminDialog.styles';
import {
  compose, get, isEmailValid, pro,
} from '../../utility';
import c from './constants';
import type { BrokerageType } from '../../flowTypes';
import { errParser } from '../../api-error-parser';

type Props = {
  classes: Object,
  closeDialog: Function,
  inviteNewAdmin: Function,
  brokerages: Array<BrokerageType>,
  onSuccessInvite: () => void,
};

type State = {
  networkError: React.Node,
};

export class InviteNewAdminDialogC extends React.Component<Props, State> {

  state = {
    networkError: '',
  };

  resetNetworkError = () => {
    this.setState({
      networkError: '',
    });
  };

  inviteNewAdmin = async ({
    email, firstName, lastName, brokerage,
  }: { email: string, firstName: string, lastName: string, brokerage: { value: string } }) => { // eslint-disable-line
    const { inviteNewAdmin, closeDialog, onSuccessInvite } = this.props;
    const trimmedEmail = email.trim();
    const [err] = await pro(inviteNewAdmin({
      email: trimmedEmail,
      firstName,
      lastName,
      brokerageId: brokerage.value,
    }));

    if (err) {
      const parsedErr = errParser.parse(err);
      if (parsedErr.network) {
        this.setState({
          networkError: c.REQUEST_ERROR,
        });
        return;
      }
      if (parsedErr.global) {
        this.setState({
          networkError: parsedErr.global,
        });
      }
    } else {
      if (onSuccessInvite) {
        onSuccessInvite();
      }
      closeDialog();
    }
  };

  render() {
    const { classes, closeDialog, brokerages } = this.props;
    const { networkError } = this.state;

    return (
      <Formik
        initialValues={{
          email: '',
          firstName: '',
          lastName: '',
          brokerage: get(brokerages[0], 'id'),
          permission: 'super-admin',
        }}
        onSubmit={this.inviteNewAdmin}
        validate={({
          email, firstName, lastName, brokerage,
        }) => {
          const errors = {};
          if (!isEmailValid(email)) {
            errors.email = c.INVALID_EMAIL;
          }
          if (!email) {
            errors.email = c.EMPTY_EMAIL;
          }
          if (!firstName) {
            errors.firstName = c.EMPTY_FIRST_NAME;
          }
          if (!lastName) {
            errors.lastName = c.EMPTY_LAST_NAME;
          }

          if (!brokerage.value) {
            errors.brokerage = c.EMPTY_BROKERAGE;
          }
          return errors;
        }}
        render={({
          values, touched, errors, handleSubmit, setFieldValue,
        }) => (
          <form onSubmit={handleSubmit}>
            <DialogTitle
              classes={{ root: classes.dialogTitle }}
              disableTypography
            >
              Invite New Admin
            </DialogTitle>
            <DialogContent classes={{ root: classes.dialogContent }}>
              <TextField
                error={touched.email && Boolean(errors.email)}
                fullWidth
                helperText={touched.email && (errors.email)}
                label="Email"
                id="email"
                InputProps={{
                  classes: {
                    root: classes.textFieldInputWrapper,
                    input: classes.textFieldInput,
                    error: classes.textFieldInputError,
                  },
                  disableUnderline: true,
                }}
                InputLabelProps={{
                  classes: { root: `${classNames(classes.textFieldLabel, classes.label)}` },
                  required: true,
                }}
                onChange={(e) => {
                  this.resetNetworkError();
                  setFieldValue('email', e.target.value);
                }}
                value={values.email}
                name="email"
              />
              <div className={classes.flexInputs}>
                <TextField
                  classes={{ root: classes.flexInput }}
                  error={touched.firstName && Boolean(errors.firstName)}
                  fullWidth
                  helperText={touched.firstName && (errors.firstName)}
                  label="First Name"
                  id="firstName"
                  InputProps={{
                    classes: {
                      root: classes.textFieldInputWrapper,
                      input: classes.textFieldInput,
                      error: classes.textFieldInputError,
                    },
                    disableUnderline: true,
                  }}
                  InputLabelProps={{
                    classes: { root: `${classNames(classes.textFieldLabel, classes.label)}` },
                    required: true,
                  }}
                  onChange={(e) => {
                    this.resetNetworkError();
                    setFieldValue('firstName', e.target.value);
                  }}
                  value={values.firstName}
                  name="firstName"
                />
                <TextField
                  classes={{ root: classes.flexInput }}
                  error={touched.lastName && Boolean(errors.lastName)}
                  fullWidth
                  helperText={touched.lastName && (errors.lastName)}
                  label="Last Name"
                  id="lastName"
                  InputProps={{
                    classes: {
                      root: classes.textFieldInputWrapper,
                      input: classes.textFieldInput,
                      error: classes.textFieldInputError,
                    },
                    disableUnderline: true,
                  }}
                  InputLabelProps={{
                    classes: { root: `${classNames(classes.textFieldLabel, classes.label)}` },
                    required: true,
                  }}
                  onChange={(e) => {
                    this.resetNetworkError();
                    setFieldValue('lastName', e.target.value);
                  }}
                  value={values.lastName}
                  name="lastName"
                />
              </div>
              <div className={classes.select}>
                <Select
                  className={classes.select}
                  options={brokerages.map(b => ({ value: b.id, label: b.brokerageName }))}
                  value={values.brokerage}
                  onChange={option => setFieldValue('brokerage', option)}
                  placeholder="To Which Brokerage Office?"
                  name="brokerage"
                />
                <FormHelperText error={touched.brokerage && Boolean(errors.brokerage)}>{touched.brokerage && errors.brokerage}</FormHelperText>
              </div>
              <FormControl component="fieldset" className={classes.radio}>
                <FormLabel component="legend" className={classNames(classes.textFieldLabel, classes.label)}>Permission</FormLabel>
                <RadioGroup
                  aria-label="Permission"
                  className={classes.radioGroup}
                  value={values.permission}
                  name="permission"
                  onChange={e => setFieldValue('permission', e.target.value)}
                >
                  <FormControlLabel value="super-admin" control={<Radio />} label="Super Admin" />
                </RadioGroup>
              </FormControl>
              <FormControl className={classes.networkErrorMessage} error={Boolean(networkError)}>
                <FormHelperText>{networkError}</FormHelperText>
              </FormControl>
            </DialogContent>
            <DialogActions
              classes={{
                root: classes.dialogActions,
                action: classes.dialogAction,
              }}
            >
              <Button variant="raised" color="default" onClick={closeDialog}>Cancel</Button>
              <Button
                variant="raised"
                color="primary"
                type="submit"
                data-testid="submit-button"
                disabled={Boolean(networkError
                  || (touched.email && errors.email)
                  || (touched.firstName && errors.firstName)
                  || (touched.lastName && errors.lastName))
                }
              >
                Invite
              </Button>
            </DialogActions>
          </form>
        )}
      />

    );
  }

}

export default compose(
  withStyles(styles, { withTheme: true }),
)(InviteNewAdminDialogC);
