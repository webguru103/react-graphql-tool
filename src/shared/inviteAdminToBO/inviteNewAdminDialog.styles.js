const styles = (theme: Object) => ({
  dialog: {
    minWidth: 500,
  },
  dialogTitle: {
    display: 'flex',
    alignItems: 'center',
    padding: '20px 24px',
    color: `${theme.palette.neutrals.mainText}`,
    fontSize: 16,
    fontWeight: 500,
    '& span': {
      flex: '1 0 0',
    },
  },
  dialogActions: {
    margin: '20px 24px',
  },
  dialogAction: {
    margin: 0,
    textTransform: 'capitalize',
    '& + $dialogAction': {
      marginLeft: 15,
    },
  },
  flexInputs: {
    display: 'flex',
    marginTop: '20px',
  },
  flexInput: {
    padding: '0 10px',
    '&:first-of-type': {
      paddingLeft: 0,
    },
    '&:last-of-type': {
      paddingRight: 0,
    },
  },
  textFieldInputWrapper: {
    'label + &': {
      marginTop: '10px',
    },
  },
  textFieldInput: {
    padding: '10px 12px',
    border: `1px solid ${theme.palette.neutrals.mainAndButtonGrey}`,
    borderRadius: 4,
    color: `${theme.palette.neutrals.mainText}`,
    fontSize: 14,
    '&:focus': {
      borderColor: `${theme.palette.primary.main}`,
    },
  },
  textFieldInputError: {
    '& $textFieldInput': {
      borderColor: '#f44336',
    },
  },
  textFieldLabel: {
    position: 'relative',
    transform: 'none',
    fontSize: 14,
  },
  label: {
    color: `${theme.palette.neutrals.mainText}`,
    '& span': {
      color: `${theme.palette.neutrals.mainText}`,
    },
  },
  select: {
    marginTop: '20px',
  },
  radio: {
    marginTop: '20px',
  },
  networkErrorMessage: {
    display: 'block',
  },
});

export default styles;
