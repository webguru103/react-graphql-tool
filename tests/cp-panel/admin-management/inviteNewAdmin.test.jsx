/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { withContext } from '../../test-helpers/contextWrapper';
import { InviteNewAdminDialogC } from '../../../src/shared/inviteAdminToBO/inviteNewAdminDialog';
import { brokerages } from './brokerages.mock';

describe('Invite New', () => {
  afterEach(cleanup);
  const Component = withContext(InviteNewAdminDialogC);
  test('renders', () => {
    const handleSubmit = jest.fn();
    const { container } = render(<Component
      classes={{}}
      inviteNewAdmin={handleSubmit}
      brokerages={brokerages}
    />);
    expect(container).toMatchSnapshot();
  });

  test('can not submit if empty fields are not filled', async (done) => {
    const handleSubmit = jest.fn();
    const { getByTestId } = render(<Component
      classes={{}}
      inviteNewAdmin={handleSubmit}
      brokerages={brokerages}
    />);
    const inviteButton = getByTestId('submit-button');

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(0);
      expect(inviteButton).toBeDisabled();
      done();
    });
  });

  test('can not submit if one field (e.g, last name) is missing', (done) => {
    const handleSubmit = jest.fn();
    const { getByLabelText, getByTestId } = render(<Component
      classes={{}}
      inviteNewAdmin={handleSubmit}
      brokerages={brokerages}
    />);
    const emailInput = getByLabelText('Email *');
    const firstNameInput = getByLabelText('First Name *');
    const inviteButton = getByTestId('submit-button');

    fireEvent.change(emailInput, {
      target: { value: 'liang@dealtap.ca' },
    });

    fireEvent.change(firstNameInput, {
      target: { value: 'Liang' },
    });

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(0);
      expect(inviteButton).toBeDisabled();
      done();
    });
  });

  test('can submit if fill all fields', (done) => {
    const handleSubmit = jest.fn(() => Promise.resolve());
    const { getByLabelText, getByTestId, getByText } = render(<Component
      classes={{}}
      inviteNewAdmin={handleSubmit}
      brokerages={brokerages}
      closeDialog={() => {}}
    />);
    const emailInput = getByLabelText('Email *');
    const firstNameInput = getByLabelText('First Name *');
    const lastNameInput = getByLabelText('Last Name *');
    const brokerageSelect = getByText('To Which Brokerage Office?');
    const inviteButton = getByTestId('submit-button');

    fireEvent.change(emailInput, {
      target: { value: 'liang@dealtap.ca' },
    });

    fireEvent.change(firstNameInput, {
      target: { value: 'Liang' },
    });

    fireEvent.change(lastNameInput, {
      target: { value: 'Cui' },
    });

    fireEvent(
      brokerageSelect,
      new MouseEvent('click', { bubbles: true }),
    );

    fireEvent.keyDown(brokerageSelect, {
      keyCode: 40,
      bubbles: true,
    });

    fireEvent.keyDown(brokerageSelect, {
      keyCode: 13,
      bubbles: true,
    });

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );

    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(1);
      done();
    });

  });
});
