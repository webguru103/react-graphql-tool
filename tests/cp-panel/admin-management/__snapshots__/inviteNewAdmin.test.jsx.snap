// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite New renders 1`] = `
<div>
  <form>
    <div
      class="MuiDialogTitle-root-1"
    >
      Invite New Admin
    </div>
    <div
      class="MuiDialogContent-root-2"
    >
      <div
        class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
      >
        <label
          class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
          data-shrink="false"
          for="email"
        >
          Email
          <span
            class="MuiFormLabel-asterisk-20"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-44 MuiInput-input-29"
            id="email"
            name="email"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <div
          class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
        >
          <label
            class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
            data-shrink="false"
            for="firstName"
          >
            First Name
            <span
              class="MuiFormLabel-asterisk-20"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-44 MuiInput-input-29"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
        >
          <label
            class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
            data-shrink="false"
            for="lastName"
          >
            Last Name
            <span
              class="MuiFormLabel-asterisk-20"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-44 MuiInput-input-29"
              id="lastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="css-10nd86i"
        >
          <div
            class="css-vj8t7z"
          >
            <div
              class="css-1hwfws3"
            >
              <div
                class="css-1492t68"
              >
                To Which Brokerage Office?
              </div>
              <div
                class="css-1g6gooi"
              >
                <div
                  class=""
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="css-1wy0on6"
            >
              <span
                class="css-d8oujb"
              />
              <div
                aria-hidden="true"
                class="css-1ep9fjw"
              >
                <svg
                  aria-hidden="true"
                  class="css-19bqh2r"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="brokerage"
            type="hidden"
            value=""
          />
        </div>
        <p
          class="MuiFormHelperText-root-51"
        />
      </div>
      <fieldset
        class="MuiFormControl-root-3"
      >
        <legend
          class="MuiFormLabel-root-14"
        >
          Permission
        </legend>
        <div
          aria-label="Permission"
          class="MuiFormGroup-root-59"
          role="radiogroup"
        >
          <label
            class="MuiFormControlLabel-root-61"
          >
            <span
              class="MuiButtonBase-root-80 MuiIconButton-root-74 MuiSwitchBase-root-70 MuiRadio-root-65 MuiRadio-colorSecondary-69 MuiSwitchBase-checked-71 MuiRadio-checked-66"
            >
              <span
                class="MuiIconButton-label-79"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-83"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked=""
                  class="MuiSwitchBase-input-73"
                  name="permission"
                  type="radio"
                  value="super-admin"
                />
              </span>
              <span
                class="MuiTouchRipple-root-92"
              />
            </span>
            <span
              class="MuiTypography-root-99 MuiTypography-body1-108 MuiFormControlLabel-label-64"
            >
              Super Admin
            </span>
          </label>
        </div>
      </fieldset>
      <div
        class="MuiFormControl-root-3"
      >
        <p
          class="MuiFormHelperText-root-51"
        />
      </div>
    </div>
    <div
      class="MuiDialogActions-root-125"
    >
      <button
        class="MuiButtonBase-root-80 MuiButton-root-127 MuiButton-contained-138 MuiButton-raised-141 MuiDialogActions-action-126"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-128"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root-92"
        />
      </button>
      <button
        class="MuiButtonBase-root-80 MuiButton-root-127 MuiButton-contained-138 MuiButton-containedPrimary-139 MuiButton-raised-141 MuiButton-raisedPrimary-142 MuiDialogActions-action-126"
        data-testid="submit-button"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-128"
        >
          Invite
        </span>
        <span
          class="MuiTouchRipple-root-92"
        />
      </button>
    </div>
  </form>
</div>
`;
